### PR TITLE
Fix/Time sync related error detection for kucoin_perpetual

### PR DESCRIPTION
**Pull request by Jbekker**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Potentially fixes this time sync related issue for kucoin_perpetual #7067.

The exception message in the logs was:
```
HTTP status is 400. Error: {"code":"400002","msg":"Invalid KC-API-TIMESTAMP"}.
```

The ``` _is_request_exception_related_to_time_synchronizer ``` method was looking for the the error code in the form of ```'ret_code <400002>'```. This did not match the above exception so the specific error handling for time sync related issues was not triggered.

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

File: hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py [hunk 1]
```diff
@@ -154,14 +154,7 @@ def start(self, clock: Clock, timestamp: float):
...
         error_description = str(request_exception)
-        ts_error_target_str = self._format_ret_code_for_print(ret_code=CONSTANTS.RET_CODE_AUTH_TIMESTAMP_ERROR)
-        param_error_target_str = (
-            "KC-API-TIMESTAMP Invalid -- Time differs from server time by more than 5 seconds"
-        )
-        is_time_synchronizer_related = (
-            ts_error_target_str in error_description or param_error_target_str in error_description
-        )
-        return is_time_synchronizer_related
+        return CONSTANTS.RET_CODE_AUTH_TIMESTAMP_ERROR in error_description and "KC-API-TIMESTAMP" in error_description
```
Comment (cardosofede) [inline hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py:157]: nice catch! did you check if also appears in kucoin spot?
If so you can add it to this PR

**Review by cardosofede — APPROVED** [review]
@rapcmia @nikspz this PR might solve some of the time sync issues that you were experiencing.

File: hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py (hunk 1 above)
Comment (Jbekker) [reply hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py:157]: Weirdly enough the method is not implemented at all there but I'm 99% sure (looking at the documentation) that the error code and message will be the same. I will add it to this PR.
I will also fix the lint issues, I see the build is failing.

Comment (cardosofede) [reply hummingbot/connector/derivative/kucoin_perpetual/kucoin_perpetual_derivative.py:157]: just approved the run

**Review by cardosofede — APPROVED** [review]

**Comment by rapcmia** [comment]
PR update:
- Test in progress and running bot on kucoin_perp

**Pull request merged**